src/extractor/guidance/intersection_generator.cpp

```diff
@@ -99,6 +99,20 @@ Intersection IntersectionGenerator::GetConnectedRoads(const NodeID from_node,
                          from_node, via_eid, traversed_in_reverse, to_node, intersection_lanes);
     };
 
+    // The first coordinate (the origin) can depend on the number of lanes turning onto,
+    // just as the target coordinate can. Here we compute the corrected coordinate for the
+    // incoming edge.
+    util::Coordinate first_coordinate;
+    double in_segment_length = 0.;
+    std::tie(first_coordinate, in_segment_length) = [&]() {
+        const auto in_coordinates =
+            coordinate_extractor.GetCoordinatesAlongRoad(from_node, via_eid, INVERT, turn_node);
+        const auto length = util::coordinate_calculation::getLength(
+            in_coordinates, util::coordinate_calculation::haversineDistance);
+        const auto coordinate = extract_coordinate(from_node, via_eid, INVERT, turn_node);
+        return std::make_pair(coordinate, length);
+    }();
```

Comment:
Not sure if the tie+lambda makes the code more readable here.. if you want to scope the code in the lambda, what about

    {   // artificially scoped
        const auto tmp = fn();
        ..
    }